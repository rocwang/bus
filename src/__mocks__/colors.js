export default {
  oceanBlue: "#001120",
  green: "#8fc240",
  blue: "#00a3da",
  yellow: "#ffdb1b",
  red: "#da2f2d",
  orange: "#ea8c2e",
  black: "#000",
  grayA: "#333",
  grayB: "#666",
  grayC: "#999",
  grayD: "#ccc",
  white: "#fff",
};
