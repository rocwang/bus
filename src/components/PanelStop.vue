<template>
  <Panel>
    <template v-slot:icon>
      <RoundIconStop />
    </template>
    <template v-slot:prefix>
      Stop
    </template>
    <template v-slot:title>{{ stopCode }}</template>
    <template v-slot:buttons>
      <Buttonizer modifier="icon">
        <button @click="$router.push({ name: 'Favourites' })" aria-label="home">
          <IconCross />
        </button>
      </Buttonizer>
    </template>
    <template v-slot:subtitle>{{ stopName }}</template>
    <template v-slot:body>
      <ul>
        <li v-for="group in routeGroups" :key="group">
          {{ group }}
        </li>
      </ul>
    </template>
  </Panel>
</template>

<script>
import Panel from "./Panel";
import RoundIconStop from "./RoundIconStop";
import IconCross from "./IconCross";
import Buttonizer from "./Buttonizer";

export default {
  name: "PanelStop",
  components: { Panel, RoundIconStop, IconCross, Buttonizer },
  props: {
    stopCode: {
      type: String,
      required: true
    },
    stopName: {
      type: String,
      required: true
    },
    routeGroups: {
      type: Array,
      required: true
    }
  },
  data() {
    return {
      stop: {}
    };
  }
};
</script>
