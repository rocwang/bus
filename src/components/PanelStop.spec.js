import { mount } from "@vue/test-utils";
import PanelStop from "./PanelStop";
import colors from "./__fixtures__/colors";

describe("PanelStop.vue", () => {
  let wrapper;
  const stopCode = "SC1234";
  const stopName = "SNXXXX";

  beforeEach(() => {
    wrapper = mount(PanelStop, {
      provide: {
        colors
      },
      propsData: {
        stopCode,
        stopName,
        routeShortNames: ["NX1", "NX2"]
      },
      stubs: ["router-link"]
    });
  });

  it("matches the snapshot", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("renders the stop code", () => {
    expect(wrapper.text()).toMatch(stopCode);
  });

  it("renders the stop name", () => {
    expect(wrapper.text()).toMatch(stopName);
  });

  it("renders the route short names ", () => {
    expect(wrapper.findAll("li").length).toBe(2);
  });

  it("has the border left color set for each route", () => {
    wrapper
      .findAll("li")
      .wrappers.forEach(li =>
        expect(li.attributes("style")).toMatch("border-left-color")
      );
  });
});
