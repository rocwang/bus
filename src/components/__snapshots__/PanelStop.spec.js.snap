// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelStop.vue matches the snapshot 1`] = `
<div
  class="root"
>
  <div
    class="head"
  >
    <div
      class="icon"
    >
      <svg
        viewBox="0 0 50 50"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M25 0a25 25 0 1 0 25 24.984C49.991 11.184 38.801 0 25 0"
          fill="#ffdb1b"
        />
         
        <path
          d="M39.063 13.75H17.5v-2.813a.94.94 0 0 0-.938-.937h-1.875a.94.94 0 0 0-.937.938v2.812h-2.813a.94.94 0 0 0-.937.938v1.874a.94.94 0 0 0 .938.938h2.812v21.563a.94.94 0 0 0 .938.937h1.874a.94.94 0 0 0 .938-.938V17.5h21.563a.94.94 0 0 0 .937-.938v-1.875a.94.94 0 0 0-.938-.937zM19.375 32.5h18.75V19.375h-18.75V32.5z"
          fill="currentColor"
          style="color: rgb(0, 17, 32);"
        />
      </svg>
    </div>
     
    <h1
      class="title"
    >
      <small
        class="prefix"
      >
        
    Stop
  
      </small>
       SC1234
    </h1>
     
    <div
      class="buttons"
    >
      <button
        aria-label="home"
        class="root icon"
      >
        <svg
          style="color: rgb(255, 219, 27);"
          viewBox="0 0 30 30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15 15L29 1 15 15 1 1l14 14zm0 0L1 29l14-14 14 14-14-14z"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </button>
    </div>
     
    <p
      class="subtitle"
    >
      SNXXXX
    </p>
  </div>
   
  <div
    class="body"
  >
    <ul
      class="routeList"
    >
      <li
        class="route"
        style="border-left-color: #8fc240;"
      >
        <router-link-stub
          class="routeLink"
          to="[object Object]"
        >
          <span
            class="routeName"
          >
            NX1
          </span>
        </router-link-stub>
      </li>
      <li
        class="route"
        style="border-left-color: #00a3da;"
      >
        <router-link-stub
          class="routeLink"
          to="[object Object]"
        >
          <span
            class="routeName"
          >
            NX2
          </span>
        </router-link-stub>
      </li>
    </ul>
  </div>
</div>
`;
